cmd/object-lambda-handlers.go

```diff
@@ -0,0 +1,338 @@
+	case "Bad Request":
+		return http.StatusBadRequest
+	case "Unauthorized":
+		return http.StatusUnauthorized
+	case "Payment Required":
+		return http.StatusPaymentRequired
+	case "Forbidden":
+		return http.StatusForbidden
+	case "Not Found":
+		return http.StatusNotFound
+	case "Method Not Allowed":
+		return http.StatusMethodNotAllowed
+	case "Not Acceptable":
+		return http.StatusNotAcceptable
+	case "Proxy Authentication Required":
+		return http.StatusProxyAuthRequired
+	case "Request Timeout":
+		return http.StatusRequestTimeout
+	case "Conflict":
+		return http.StatusConflict
+	case "Gone":
+		return http.StatusGone
+	case "Length Required":
+		return http.StatusLengthRequired
+	case "Precondition Failed":
+		return http.StatusPreconditionFailed
+	case "Request Entity Too Large":
+		return http.StatusRequestEntityTooLarge
+	case "Request URI Too Long":
+		return http.StatusRequestURITooLong
+	case "Unsupported Media Type":
+		return http.StatusUnsupportedMediaType
+	case "Requested Range Not Satisfiable":
+		return http.StatusRequestedRangeNotSatisfiable
+	case "Expectation Failed":
+		return http.StatusExpectationFailed
+	case "I'm a teapot":
+		return http.StatusTeapot
+	case "Misdirected Request":
+		return http.StatusMisdirectedRequest
+	case "Unprocessable Entity":
+		return http.StatusUnprocessableEntity
+	case "Locked":
+		return http.StatusLocked
+	case "Failed Dependency":
+		return http.StatusFailedDependency
+	case "Too Early":
+		return http.StatusTooEarly
+	case "Upgrade Required":
+		return http.StatusUpgradeRequired
+	case "Precondition Required":
+		return http.StatusPreconditionRequired
+	case "Too Many Requests":
+		return http.StatusTooManyRequests
+	case "Request Header Fields Too Large":
+		return http.StatusRequestHeaderFieldsTooLarge
+	case "Unavailable For Legal Reasons":
+		return http.StatusUnavailableForLegalReasons
+	case "Internal Server Error":
+		return http.StatusInternalServerError
+	case "Not Implemented":
+		return http.StatusNotImplemented
+	case "Bad Gateway":
+		return http.StatusBadGateway
+	case "Service Unavailable":
+		return http.StatusServiceUnavailable
+	case "Gateway Timeout":
+		return http.StatusGatewayTimeout
+	case "HTTP Version Not Supported":
+		return http.StatusHTTPVersionNotSupported
+	case "Variant Also Negotiates":
+		return http.StatusVariantAlsoNegotiates
+	case "Insufficient Storage":
+		return http.StatusInsufficientStorage
+	case "Loop Detected":
+		return http.StatusLoopDetected
+	case "Not Extended":
+		return http.StatusNotExtended
+	case "Network Authentication Required":
+		return http.StatusNetworkAuthenticationRequired
+	default:
+		return -1
+	}
+}
+
+func fwdHeadersToS3(h http.Header, w http.ResponseWriter) {
+	for k, v := range h {
+		if !strings.HasPrefix(strings.ToLower(k), "x-amz-fwd-header-") {
+			continue
+		}
+		nk := strings.TrimPrefix(k, "x-amz-fwd-header-")
+		nk = strings.TrimPrefix(nk, "X-Amz-Fwd-Header-")
+		w.Header()[nk] = v
+	}
+}
```

Comment:
```suggestion
func fwdHeadersToS3(h http.Header, w http.ResponseWriter) {
	const trim = "x-amz-fwd-header-"
	for k, v := range h {
		if !strings.HasPrefix(strings.ToLower(k), trim) {
			continue
		}
		w.Header()[k[len(trim):]] = v
	}
}
```

Reply:
We have to preserve certain headers as is to not make it canonical always @klauspost - AWS SDKs are stupid some times and expect case sensitivity.

Reply:
@harshavardhana ok, updated the proposal.

Reply:
it may still not work @klauspost because Go converts non-canonical headers into canonical ones :-) @klauspost 

that's why I have that kind of code for both forms.

Reply:
and I cant manually convert `k` to a lower-case form, because I need to preserve the case sensitivity after the trim.

Reply:
@harshavardhana Look again. It does exactly the same as yours :) (and it keeps whatever case is there)

Reply:
how the trim is lowercased? here @klauspost I don't know the case for incoming `k` in `http.Header` I need to try both lowercased as well as camelcased.

Reply:
@harshavardhana `strings.HasPrefix(strings.ToLower(k), trim)` will pass through any case of `X-AmZ-FWd-HeaDer-`. 

Since the length is the same no matter the case, we trim that number of characters from the front.

Another way would be:

```suggestion
func fwdHeadersToS3(h http.Header, w http.ResponseWriter) {
	const trim = "x-amz-fwd-header-"
	for k, v := range h {
		if strings.HasPrefix(strings.ToLower(k), trim) {
			w.Header()[k[len(trim):]] = v
		}
	}
}
```

Reply:
yes but this doesn't trim when `k` has `X-Amz-Fwd-Header-Content-Type` @klauspost so we are only trimming lowercase. 

That's why I have code to do double 'trim' once with `x-amz-fwd-header-` and `X-Amz-Fwd-Header`

Reply:
I made the change and tested it.